pkg/etcd/etcd.go

```diff
@@ -854,3 +854,82 @@ func backupDirWithRetention(dir string, maxBackupRetention int) (string, error)
 	}
 	return backupDir, nil
 }
+
+func GetServerURLFromETCD(ctx context.Context, cfg *config.Control) (string, int, error) {
+	if cfg.Runtime == nil {
+		return "", 0, fmt.Errorf("runtime is not ready yet")
+	}
+	cl, err := getClient(ctx, cfg.Runtime, endpoint)
+	if err != nil {
+		return "", 0, err
+	}
+	etcdResp, err := cl.KV.Get(ctx, serverKey)
+	if err != nil {
+		return "", 0, err
+	}
+
+	if etcdResp.Count < 1 {
+		return "", 0, fmt.Errorf("servers urls are not yet set")
+	}
+	serverURL := strings.TrimSuffix(string(etcdResp.Kvs[0].Value), "\n")
+	serverURL = serverURL[1 : len(serverURL)-1]
+
+	serverAddress := strings.Split(serverURL, ":")
+	port, err := strconv.Atoi(serverAddress[1])
+	if err != nil {
+		return "", 0, fmt.Errorf("port is not set correctly: %v", err)
+	}
```

Comment:
Save a bunch of splitting and string converting, and just have the etcd key store a comma-separated list of server URLs since that's what we end up converting this to anyways.